sale_configurator_option/models/sale.py

```diff
@@ -6,6 +6,23 @@
 from odoo import api, fields, models
 
 
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def copy_data(self, default=None):
+        # Option lines should not be copied directly but from parent line option_ids
+        if default is None:
+            default = {}
+        if "order_line" not in default:
+            default["order_line"] = [
+                (0, 0, line.copy_data()[0])
+                for line in self.order_line.filtered(
+                    lambda l: not l.is_downpayment and not l.parent_id
```

Comment:
for the version 14 we can make better then hacking this.
On the futur v18 it will better if we can improve the native code

https://github.com/odoo/odoo/blob/01b14734a9094ca1d94fbbdc682d93cfc7042764/addons/sale/models/sale_order.py#L805

Having someting like

```
            default["order_line"] = [
                (0, 0, line.copy_data()[0])
                for line in self.order_line.filtered(
                    lambda l: l.should_be_copied()
                )
            ]



def should_be_copied(self):
    self.ensure_one()
    return not self.is_downpayment

```

So we can just inherit the method "should_be_copied" 


Can you do a PR on odoo master for this ?

Thanks

Reply:
https://github.com/odoo/odoo/pull/170873